### PR TITLE
[#19303] fix: edit profile design feedbacks

### LINKED ISSUE #19303: Profile - Edit profile screen issues
# Design Feedback 2.27

## [Profile] Edit profile screen issues
Figma frame of the Review [here](https://www.figma.com/file/oznddnBnDbLlLQIaACYNuj/Release-Reviews?type=design&node-id=36%3A115879&mode=design&t=O3UNQJbRpr0lqKTe-1)

- **Wrong error message** [[comment](https://www.figma.com/file/oznddnBnDbLlLQIaACYNuj?node-id=47:31015&mode=design#716141545)] when using an apostrophe in the profile bio
- **Allowed symbols in bio** [[comment](https://www.figma.com/file/oznddnBnDbLlLQIaACYNuj?node-id=47:31015&mode=design#716141545)] should include apostrophes, commas, dots and dashes

Screenshot 2024-03-18 alle 10 25 29

>
- **Photo button** [[comment](https://www.figma.com/file/oznddnBnDbLlLQIaACYNuj?node-id=36:115879&mode=design#727735372)] shouldn't be a circle
- **Toast copy** [[comment](https://www.figma.com/file/oznddnBnDbLlLQIaACYNuj?node-id=56:35209&mode=design#719297061)] if the profile picture was changed (not added for the first time) the toast should say _"Profile picture updated"_ not "added". This also applies to bio, custom color, etc.

Screenshot 2024-03-18 alle 10 25 56

### Reproduction

1. Open profile settings by tapping on the avatar picture at the top left corner of the screen
2. Tap on _Bio_ to edit a pre existent bio
3. Save changes

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS
- Devices in the screenshots: iPhone 13

[comment]: # (Please, add logs/notes if necessary)

### LINKED ISSUE #19326: Special characters entered into Bio field not matching with Desktop
Related to question https://discord.com/channels/624634427930312714/928625369542713396/1207020237304893478

### Steps:
1. Go to profile
2. Edit profile
3. Open bio option
4. Do the following actions:
- Tap enter to start a new line
- Enter dot, comma or other characters

### Actual result:
The 'special characters are not allowed' validation is shown

https://github.com/status-im/status-mobile/assets/52490791/b21f17d9-a22c-4497-ac1f-9ca4931bc3d8

### Note:
Input from the design team is needed about the exact characters allowed to enter into the bio field to match with desktop.

### PR DESCRIPTION
**Pull request by mohsen-ghafouri**
fixes #19303
fixes #19326

### Summary

Design feedbacks in Edit profile screen

#### Areas that maybe impacted

- Edit Bio
- Change profile pic

### Result

https://github.com/status-im/status-mobile/assets/71308738/3099d5ac-3fa6-4c91-8e18-ffe3a577c989

status: ready

### CODE REVIEW

**Review by jo-mut — APPROVED** [review]

**Comment by mohsen-ghafouri** [comment]
@status-im/mobile-qa could you please check test results and move it to design review?

**Comment by VolodLytvynenko** [comment]
I have rerun 1 id: 702844. 'Device 1' sends a message with a link to 'Device 2', but 'Device 2' doesn't receive the link. Seems that delivery issue. Lets see rerun result

**Comment by mohsen-ghafouri** [comment]
Hey @VolodLytvynenko, sorry this PR requires manual QA as we changed the validation rules of bio. this is our reference for rules https://www.figma.com/file/ibJOTPlNtIxESwS96vJb06/%F0%9F%91%A4-Profile-%7C-Desktop?type=design&node-id=4621-461562&mode=design&t=nGT7ParxL7vVk7qh-4

**Comment by VolodLytvynenko** [comment]
> Hey @VolodLytvynenko, sorry this PR requires manual QA as we changed the validation rules of bio. this is our reference for rules https://www.figma.com/file/ibJOTPlNtIxESwS96vJb06/%F0%9F%91%A4-Profile-%7C-Desktop?type=design&node-id=4621-461562&mode=design&t=nGT7ParxL7vVk7qh-4

@mohsen-ghafouri thank you for update. I will take a look

**Comment by VolodLytvynenko** [comment]
Hi @Francesca-G, the design [here](https://www.figma.com/file/oznddnBnDbLlLQIaACYNuj/Release-Reviews?type=design&node-id=36-115879&mode=design&t=Xma7L313AEgGM3ZL-0#716141545) described that apostrophes, commas, dots, and dashes should be allowed in descriptions. Should we only allow these special characters, or should we match the desktop's behavior, which allows all characters? Currently, the desktop allows all characters

that's why I created an issue with the '👀 design-input-required' label [here](https://github.com/status-im/status-mobile/issues/19326) to get the correct specification, to avoid conflicts during syncing between mobile devices and desktop. 

For example:
1. A desktop user sets all possible characters in the bio.
2. The mobile device syncs with the desktop

**Comment by VolodLytvynenko** [comment]
Hi @mohsen-ghafouri. thank you for PR. Take a look at the found issue:

## PR_ISSUE 1: [Android] 'maximum regex stach depth reached' error is shown after entering special characters and then non-allowed characters

### Steps:
1. Go to the bio 
2. Enter special characters "apostrophes, commas, dots" -> save
3. Reopen bio input field
4. Enter non-allowed characters
5. Wait at least 2 minutes

### Actual result:
image

### Expected result:
No errors are shown
### OS: 
Android

### Devices:
- Pixel 7a, Android 13

### Logs
[Status-debug-logs.zip](https://github.com/status-im/status-mobile/files/14977007/Status-debug-logs.zip)

**Comment by Francesca-G** [comment]
> Hi @Francesca-G, the design [here](https://www.figma.com/file/oznddnBnDbLlLQIaACYNuj/Release-Reviews?type=design&node-id=36-115879&mode=design&t=Xma7L313AEgGM3ZL-0#716141545) described that apostrophes, commas, dots, and dashes should be allowed in descriptions. Should we only allow these special characters, or should we match the desktop's behavior, which allows all characters? Currently, the desktop allows all characters
> 

I think it makes sense to match desktop's behavior like you suggested 👍

**Comment by VolodLytvynenko** [comment]
@Francesca-G In such case all validation messages on figma created for bio filed are redundant. The BIO field should accept all characters including emojis as it works on desktop. Is it correct?

File: src/status_im/common/validation/profile.cljs [hunk 1]
```diff
@@ -9,6 +9,9 @@
 ;; https://github.com/status-im/status-desktop/blob/2ba96803168461088346bf5030df750cb226df4c/ui/imports/utils/Constants.qml#L468
 (def min-length 5)
 
+(def bio-chars-regex
+  #"^(\u00a9|\u00ae|[\u2000-\u3300]|\ud83c[\ud000-\udfff]|\ud83d[\ud000-\udfff]|\ud83e[\ud000-\udfff]|[\u0000-\u007F]+)+$")
```
Comment (OmarBasem) [inline src/status_im/common/validation/profile.cljs]: Maybe add a comment/docstring about what this regex does?

**Review by OmarBasem — APPROVED** [review]

**Comment by mohsen-ghafouri** [comment]
@Francesca-G @VolodLytvynenko as we discussed with @benjthayer and Desktop team this is our latest agreement https://www.figma.com/file/ibJOTPlNtIxESwS96vJb06/%F0%9F%91%A4-Profile-%7C-Desktop?type=design&node-id=4621-461562&mode=design&t=QImX3GeiQcXELSFG-0 and the only validation is for not accepting UTF-8 chars, design team suppose to update mobile ui as well.

**Comment by VolodLytvynenko** [comment]
> @Francesca-G @VolodLytvynenko as we discussed with @benjthayer and Desktop team this is our latest agreement https://www.figma.com/file/ibJOTPlNtIxESwS96vJb06/%F0%9F%91%A4-Profile-%7C-Desktop?type=design&node-id=4621-461562&mode=design&t=QImX3GeiQcXELSFG-0 and the only validation is for not accepting UTF-8 chars, design team suppose to update mobile ui as well.

Hi @mohsen-ghafouri thank you for clarification!

**Comment by mohsen-ghafouri** [comment]
Hi @VolodLytvynenko, I couldn't reproduce the issue exactly but from the log I applied some changes, could you please recheck?

**Comment by VolodLytvynenko** [comment]
Hi @mohsen-ghafouri thank you for fixes. [Issue 1](https://github.com/status-im/status-mobile/pull/19562#issuecomment-2056403455) is fixed 

Unfortunately, additional issues are found. Take a look please

## PR_ISSUE 2: Unable to save max available value of ASCII characters in Bio

### Steps:
1. Enter random ASCII characters of 240 length into the bio. (example: [chars.txt](https://github.com/status-im/status-mobile/files/15007617/chars.txt)
)
2. Save

### Actual result:
The save button does not respond.

https://github.com/status-im/status-mobile/assets/52490791/9740fe77-00df-4d55-98ee-2e3e60927836


### Expected result:
The 240 characters should be saved successfully.

### OS: 
IOS, Android

### Devices:
- Pixel 7a, Android 13
- iPhone 11 Pro Max, IOS 17
### Logs 
[Status-debug-logs.zip](https://github.com/status-im/status-mobile/files/15007718/Status-debug-logs.zip)

**Comment by VolodLytvynenko** [comment]
Unfortunately, there are no exact steps to reproduce this issue, seems it happens randomly. I hope the logs might help understand what's going on

## PR_ISSUE 3: [Android] The bio page can't be closed

### Steps:
1. Enter valid data into bio
2. tap save or back button to close the bio page


### Actual result:
Bio page can't be closed

https://github.com/status-im/status-mobile/assets/52490791/18f15b24-0250-40c6-b703-0c2dc8f6da73



### Expected result:
Bio page is closed when save or back button are tapped

### OS: 
Android

### Devices:
- Pixel 7a, Android 13

### Logs
[Status-debug-logs (1).zip](https://github.com/status-im/status-mobile/files/15007876/Status-debug-logs.1.zip)

**Comment by VolodLytvynenko** [comment]
hi @mohsen-ghafouri could you also change the validation message in scope of this PR, recently it was changed in Figma?  It should now be 'Invalid characters. Please use only standard keyboard characters and emojis'
reference is here: https://github.com/status-im/status-mobile/issues/19326#issuecomment-2063570122

**Comment by mohsen-ghafouri** [comment]
Hi @VolodLytvynenko, sure will do

**Comment by mohsen-ghafouri** [comment]
@VolodLytvynenko could you please check again? bio length should be fine now. also for second issue i saw many waku logs and couldn't reproduce the issue. please test again with latest changes. thanks

**Comment by VolodLytvynenko** [comment]
Hi @mohsen-ghafouri could you please rebase current and status go PR https://github.com/status-im/status-go/tree/fix/set-bio-length. Thank you

**Comment by mohsen-ghafouri** [comment]
Hi @VolodLytvynenko, done.

**Comment by VolodLytvynenko** [comment]
Hi @mohsen-ghafouri Thank you for the fixes. No issues from my side. PR is ready to be merged

**Pull request merged**